### PR TITLE
Fixes VS Build Drop reference in OptProfV2

### LINKED ISSUE #8671: Support for OptProfV2 CI runs in Escrow mode
In https://github.com/NuGet/Home/issues/8578 , we disabled OptProfV2 related CI tasks to unblock the NuGet 5.3 release on VS. Now, we want to re-enable them to provide training data for 16.4 release, even when escrow mode is on course.

### PR DESCRIPTION
**Pull request by dominoFire**
## Bug

Fixes: https://github.com/NuGet/Home/issues/8671
Regression: Yes
* Last working version: Last good build is 5.3.0.6234
* How are we preventing it in future: Thoughtful review of arcade documentation.

## Fix

-  Removes dependency on $ShouldSkipOptimize variable: all related OptProfV2 tasks will run even if the plugin installation is disabled
- Corrected VS Drop reference

## Testing/Validation

Tests Added: No
Reason for not adding tests: CI infrastructure, needs an official build.
Validation: Local dotnet/arcade execution (in progress)

### CODE REVIEW

File: build/vsts_build.yaml [hunk 1]
```diff
@@ -345,14 +345,14 @@ phases:
     inputs:
       command: 'custom'
       arguments: 'sources add -Name VS -Source $(VsPackageFeedUrl) -UserName $(VsPackageFeedUsername) -Password $(VsPackageFeedPassword) -ConfigFile $(System.DefaultWorkingDirectory)\NuGet.config'
-    condition: "and(succeeded(), eq(variables['BuildRTM'], 'false'), eq(variables['IsOfficialBuild'], 'true'), eq(variables['ShouldSkipOptimize'], 'false'))"
+    condition: "and(succeeded(), eq(variables['BuildRTM'], 'false'), eq(variables['IsOfficialBuild'], 'true'))"
```
Comment (nkolev92) [inline build/vsts_build.yaml:348]: Why are we removing these? 

Aren't they're useful for disabling again if we have to?

File: build/config.props [hunk 2]
```diff
@@ -33,7 +33,8 @@
     <VsTargetBranch>master</VsTargetBranch>
     <VsTargetChannel>int.$(VsTargetBranch)</VsTargetChannel>
     <VsTargetBranch Condition="'$(IsEscrowMode)' == 'true'">rel/d$(VsTargetMajorVersion).$(MinorNuGetVersion)</VsTargetBranch>
-    <VsTargetChannel Condition="'$(IsEscrowMode)' == 'true'">Preview</VsTargetChannel>
+    <!-- VS short branch name. See https://github.com/dotnet/arcade/blob/master/Documentation/ArcadeSdk.md#visual-studio-training-inputs-generation -->
```
Comment (nkolev92) [inline build/config.props]: I'm not sure what arcade has to do with this. 
We are not onboarded to arcade

File: build/vsts_build.yaml (hunk 1 above)
Comment (nkolev92) [reply build/vsts_build.yaml:348]: Why should they always run?

File: build/config.props [hunk 3]
```diff
@@ -33,7 +33,8 @@
...
+    <VsTargetChannel Condition="'$(IsEscrowMode)' == 'true'">int.d$(VsTargetMajorVersion).$(MinorNuGetVersion)</VsTargetChannel>
```
Comment (nkolev92) [inline build/config.props:36]: Can you elaborate on this?

File: build/vsts_build.yaml (hunk 1 above)
Comment (dtivel) [reply build/vsts_build.yaml:348]: The clause should not have been added in the [first place](https://github.com/NuGet/NuGet.Client/pull/3057).

These OptProfV2 tasks are not always run.  They are only run on release branches if the build is successful so far.  Task execution should not be conditioned on the value of `ShouldSkipOptimize`.  The purpose of `ShouldSkipOptimize` is to temporarily disable optimization (when `ShouldSkipOptimize` is `true`) so as to unblock the build and allow new optimization profiling data to be created through these OptProfV2 tasks.  Then, `ShouldSkipOptimize` should be reverted to the default value of `false`.

As long as optimization profiling data exists, there should be no need to toggle `ShouldSkipOptimize`.

File: build/config.props (hunk 2 above)
Comment (dtivel) [reply build/config.props]: I would avoid this link and just inline a distilled explanation.

**Review by dtivel — APPROVED** [review]

File: build/config.props (hunk 2 above)
Comment (dominoFire) [reply build/config.props]: That was the most recent documentation I found about VS Channel names for OptProf. I will remove the link and add documentation in the team notebook

File: build/config.props (hunk 3 above)
Comment (dominoFire) [reply build/config.props:36]: It's now on the team notebook

File: build/vsts_build.yaml (hunk 1 above)
Comment (dominoFire) [reply build/vsts_build.yaml:348]: Agree with Damon. The idea is to always generate training data, even when we're not creating optimized binaries (when ShouldSkipOptimize = true)

**Review by donnie-msft — APPROVED** [review]

**Review by nkolev92 — APPROVED** [review]

**Review by zivkan — APPROVED** [review]

**Pull request merged**